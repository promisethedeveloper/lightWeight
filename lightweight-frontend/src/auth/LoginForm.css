.LoginForm-heading {
	color: #461111;
}

label {
	color: #a13333;
}

.LoginForm-button {
	color: white;
	background-color: #461111;
}

button:hover {
	background-color: #a13333;
	color: white;
}
