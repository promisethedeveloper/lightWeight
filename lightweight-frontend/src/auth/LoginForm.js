import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import Alert from "../common/Alert";
import "./LoginForm.css";

const LoginForm = ({ login }) => {
	const history = useHistory();
	const [formData, setFormData] = useState({
		username: "",
		password: "",
	});
	const [formErrors, setFormErrors] = useState([]);

	/**
	 * Handle form submit:
	 *
	 * Calls login function prop and if successful, redirect to "/"
	 */

	const handleSubmit = async (e) => {
		e.preventDefault();
		let result = await login(formData);
		if (result.success) {
			history.push("/");
		} else {
			setFormErrors(result.errors);
		}
	};

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((data) => ({ ...data, [name]: value }));
	};

	return (
		<div className="LoginForm">
			<div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
				<h3 className="mb-3 LoginForm-heading">Log In</h3>

				<div className="card">
					<div className="card-body">
						<form onSubmit={handleSubmit}>
							<div className="form-group">
								<label htmlFor="username">Username</label>
								<input
									id="username"
									name="username"
									className="form-control"
									value={formData.username}
									onChange={handleChange}
									autoComplete="username"
									required
								/>
							</div>
							<div className="form-group">
								<label htmlFor="password">Password</label>
								<input
									id="password"
									type="password"
									name="password"
									className="form-control"
									value={formData.password}
									onChange={handleChange}
									autoComplete="current-password"
									required
								/>
							</div>
							{formErrors.length ? (
								<Alert type="danger" messages={formErrors} />
							) : null}
							<button className="btn float-right font-weight-bold LoginForm-button">
								LOGIN
							</button>
						</form>
					</div>
				</div>
			</div>
		</div>
	);
};

export default LoginForm;
