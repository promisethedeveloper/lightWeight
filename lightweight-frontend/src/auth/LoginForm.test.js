import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import LoginForm from "./LoginForm";

/** Smoke test */
it("renders without crashing", () => {
	render(
		<MemoryRouter>
			<LoginForm />
		</MemoryRouter>
	);
});

/** Snapshot test */
it("Should match snapshots", () => {
	const { asFragment } = render(<LoginForm />);
	expect(asFragment()).toMatchSnapshot();
});

it("renders a 'Log In' text", () => {
	const { getByText } = render(<LoginForm />);
	expect(getByText("Log In")).toBeInTheDocument();
});

it("renders form's username and password label text", () => {
	const { getByLabelText } = render(<LoginForm />);
	expect(getByLabelText("Username")).toBeInTheDocument();
	expect(getByLabelText("Password")).toBeInTheDocument();
});
