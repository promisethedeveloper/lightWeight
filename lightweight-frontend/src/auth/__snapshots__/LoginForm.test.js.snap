// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshots 1`] = `
<DocumentFragment>
  <div
    class="LoginForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h3
        class="mb-3 LoginForm-heading"
      >
        Log In
      </h3>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                for="username"
              >
                Username
              </label>
              <input
                autocomplete="username"
                class="form-control"
                id="username"
                name="username"
                required=""
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="current-password"
                class="form-control"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
            <button
              class="btn float-right font-weight-bold LoginForm-button"
            >
              LOGIN
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
